tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluation.scala

```diff
@@ -0,0 +1,148 @@
+package org.allenai.nlpstack.parse.poly.eval
+
+import org.allenai.nlpstack.parse.poly.polyparser._
+
+object ParseEvaluation {
+
+  /** Runs a specified scoring function over all parses in the parse source, and sums the results.
+    *
+    * Each scoring function returns a ratio (i.e. numerator/denominator) for each parse in
+    * the parse source. The sum of these ratios is returned.
+    *
+    * @param scoringFunction the scoring function to apply to the parses
+    * @param parseSource the source of parses
+    * @return the sum of the ratios produced by calling the scoring function on each parse of the
+    * parse source
+    */
+  def scoreParseSource(
+    scoringFunction: ParseScore,
+    parseSource: PolytreeParseSource
+  ): (Int, Int) = {
+
+    (parseSource.parseIterator map { candidateParse =>
+      scoringFunction.getRatio(candidateParse)
+    }) reduce { (x, y) => (x._1 + y._1, x._2 + y._2) }
```

Comment:
So what this is doing is the following:

For the first parse, it'll find the raw fraction of correct events over total events. So like 5 correctly attached tokens out of 8 overall tokens.

For the second parse, it'll again find the raw fraction of correct events over total events. Say 4 correctly attached tokens out of 7 overall tokens.

To accumulate these, we want to get the result: 9 correctly attached tokens out of 15 overall tokens.